repo_refactoring/code/model.py

```diff
@@ -114,11 +133,30 @@ def write_log(log_file):
 
 class Toy_model(Model):
     def _train(self, df_train, df_valid):
+        # train a simple random forest (RF)
+        # unlike xgboost, 
+        # for RF, we don't need a validation set for training
+        from sklearn.ensemble import RandomForestClassifier
+        # prepare the data
+        y = df_train['target']
+        df_train.drop('target', axis=1, inplace=True)
+        X = df_train
+        # prepare the model
+        model_param = {}
+        for key in self._param:
+            if key not in ['features', 'random_state']:
+                model_param[key] = self._param[key]
```

Comment:
This loop works, but you could consider something like
model_param = { k : v for (k, v) in self._param.item() if k not in ['features', 'random_state'] }

And I was wondering can we just pass in those useless values without any filtering

Reply:
I just tested, it won't work... it will give '... got an unexpected keyword argument ...' error